src/arch/src/x86_64/msr.rs

```diff
@@ -0,0 +1,279 @@
+        MSR_PKGC3_IRTL,
+        3
+    ),
+    SINGLE_MSR!(MSR_PKG_POWER_LIMIT),
+    SINGLE_MSR!(MSR_PKG_ENERGY_STATUS),
+    SINGLE_MSR!(MSR_PKG_PERF_STATUS),
+    SINGLE_MSR!(MSR_PKG_POWER_INFO),
+    SINGLE_MSR!(MSR_DRAM_POWER_LIMIT),
+    SINGLE_MSR!(MSR_DRAM_ENERGY_STATUS),
+    SINGLE_MSR!(MSR_DRAM_PERF_STATUS),
+    SINGLE_MSR!(MSR_DRAM_POWER_INFO),
+    SINGLE_MSR!(MSR_CONFIG_TDP_NOMINAL),
+    SINGLE_MSR!(MSR_CONFIG_TDP_LEVEL_1),
+    SINGLE_MSR!(MSR_CONFIG_TDP_LEVEL_2),
+    SINGLE_MSR!(MSR_CONFIG_TDP_CONTROL),
+    SINGLE_MSR!(MSR_TURBO_ACTIVATION_RATIO),
+    SINGLE_MSR!(MSR_IA32_TSCDEADLINE),
+    MSR_RANGE!(APIC_BASE_MSR, APIC_MSR_INDEXES),
+    SINGLE_MSR!(MSR_IA32_BNDCFGS),
+    SINGLE_MSR!(MSR_KVM_WALL_CLOCK_NEW),
+    SINGLE_MSR!(MSR_KVM_SYSTEM_TIME_NEW),
+    SINGLE_MSR!(MSR_KVM_ASYNC_PF_EN),
+    SINGLE_MSR!(MSR_KVM_STEAL_TIME),
+    SINGLE_MSR!(MSR_KVM_PV_EOI_EN),
+    SINGLE_MSR!(MSR_EFER),
+    SINGLE_MSR!(MSR_STAR),
+    SINGLE_MSR!(MSR_LSTAR),
+    SINGLE_MSR!(MSR_CSTAR),
+    SINGLE_MSR!(MSR_SYSCALL_MASK),
+    SINGLE_MSR!(MSR_FS_BASE),
+    SINGLE_MSR!(MSR_GS_BASE),
+    SINGLE_MSR!(MSR_KERNEL_GS_BASE),
+    SINGLE_MSR!(MSR_TSC_AUX),
+];
+
+/// Specifies whether a particular MSR should be included in vcpu serialization.
+///
+/// # Arguments
+///
+/// * `index` - The index of the MSR that is checked whether it's needed for serialization.
+pub fn msr_should_serialize(index: u32) -> bool {
+    // Blacklisted MSRs not exported by Linux: IA32_FEATURE_CONTROL and IA32_MCG_CTL
+    if index == MSR_IA32_FEATURE_CONTROL || index == MSR_IA32_MCG_CTL {
+        return false;
+    };
+    for range in WHITELISTED_MSR_RANGES.iter() {
+        if range.contains(index) {
+            return true;
+        }
+    }
+    false
+}
+
+// Creates and populates required MSR entries for booting Linux on X86_64.
+fn create_boot_msr_entries() -> Vec<kvm_msr_entry> {
+    let msr_entry_default = |msr| kvm_msr_entry {
+        index: msr,
+        data: 0x0,
+        ..Default::default()
+    };
+
+    vec![
+        msr_entry_default(MSR_IA32_SYSENTER_CS),
+        msr_entry_default(MSR_IA32_SYSENTER_ESP),
+        msr_entry_default(MSR_IA32_SYSENTER_EIP),
+        // x86_64 specific msrs, we only run on x86_64 not x86.
+        msr_entry_default(MSR_STAR),
+        msr_entry_default(MSR_CSTAR),
+        msr_entry_default(MSR_KERNEL_GS_BASE),
+        msr_entry_default(MSR_SYSCALL_MASK),
+        msr_entry_default(MSR_LSTAR),
+        // end of x86_64 specific code
+        msr_entry_default(MSR_IA32_TSC),
+        kvm_msr_entry {
+            index: MSR_IA32_MISC_ENABLE,
+            data: u64::from(MSR_IA32_MISC_ENABLE_FAST_STRING),
+            ..Default::default()
+        },
+    ]
+}
+
+/// Configure Model Specific Registers (MSRs) required to boot Linux for a given X86_64 vCPU.
+///
+/// # Arguments
+///
+/// * `vcpu` - Structure for the VCPU that holds the VCPU's fd.
+pub fn setup_msrs_for_boot(vcpu: &VcpuFd) -> Result<()> {
```

Comment:
nit: `setup_msrs` is shorter and more consistent with the other methods:
`setup_regs`
`setup_fpu`
`setup_sregs`